### PR TITLE
new tool snap-aligner/index

### LINKED ISSUE #1501: new module: snap-aligner
I think it would be good to have a module for [snap-aligner](https://bioconda.github.io/recipes/snap-aligner/README.html)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #1501 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/snapaligner/index/main.nf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    script:
+    def args = task.ext.args ?: ''
+    """
+    mkdir snap
+    snap-aligner \\
+    index \\
+    $fasta \\
+    snap \\
+    -t${task.cpus} \\
+    $args
```
Comment (jfy133) [inline modules/snapaligner/index/main.nf:31]:
```suggestion
      index \\
      $fasta \\
      snap \\
      -t${task.cpus} \\
      $args
```

Formatting to make Harshil happy ;)

File: modules/snapaligner/index/main.nf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    path "snap/*"            ,emit: index
+    path "versions.yml"      ,emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    mkdir snap
```
Comment (jfy133) [inline modules/snapaligner/index/main.nf:25]:
```suggestion
    mkdir snap
    
```

File: modules/snapaligner/index/main.nf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    conda (params.enable_conda ? "bioconda::snap-aligner=2.0.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/snap-aligner:2.0.1--hd03093a_1':
+        'quay.io/biocontainers/snap-aligner:2.0.1--hd03093a_1' }"
+
+    input:
+    path fasta
+    path altcontigfile
+    path nonaltcontigfile
```
Comment (jfy133) [inline modules/snapaligner/index/main.nf:13]: missing support fo `-altLiftoverFile`

File: modules/snapaligner/index/main.nf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    snap-aligner \\
+    index \\
+    $fasta \\
+    snap \\
+    -t${task.cpus} \\
+    $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        snapaligner: \$(snap-aligner 2>&1| head -n 1 | sed 's/^.*version //')
```
Comment (jfy133) [inline modules/snapaligner/index/main.nf:35]: This leaves an extra trailing `.` you need to remove

```
❯ snap-aligner 2>&1| head -n 1 | sed 's/^.*version //'
2.0.1.
```

**Review by jfy133 — CHANGES_REQUESTED** [review]

File: modules/snapaligner/index/main.nf (hunk 4 above)
Comment (matthdsm) [reply modules/snapaligner/index/main.nf:43]: should be fixed

File: modules/snapaligner/index/main.nf (hunk 3 above)
Comment (matthdsm) [reply modules/snapaligner/index/main.nf:13]: fixed

File: modules/snapaligner/index/main.nf (hunk 1 above)
Comment (matthdsm) [reply modules/snapaligner/index/main.nf]: The editorconf linter doesn't seem to like your indentations 😅

Comment (jfy133) [reply modules/snapaligner/index/main.nf]: oops might have been four I forget..

**Review by jfy133 — APPROVED** [review]

**Pull request merged**